tests/utils/test_cache_utils.py

```diff
@@ -916,10 +901,167 @@ def test_hybrid_cache_prompt_longer_than_max_cache_len(self):
             layer_idx=0,
             cache_kwargs={"cache_position": cache_position, "sliding_window": self.window_size},
         )
-        stored_keys, stored_values = self._extract_both_caches(hybrid_cache)
+        k = hybrid_cache.key_cache[0][0, 0, :, 0].tolist()
+        v = hybrid_cache.value_cache[0][0, 0, :, 0].tolist()
+        self.assertEqual(k, values[-self.window_size :], "HybridCache (sliding) did not keep last window tokens")
+        self.assertEqual(v, values[-self.window_size :], "HybridCache (sliding) did not keep last window tokens")
+
+    def test_static_cache_hardcoded(self):
+        """Test StaticCache with manually prefilled states and hardcoded assertions."""
+        # Scenario 1: Fill up to near capacity
+        static_cache = StaticCache(config=self.config, max_batch_size=1, max_cache_len=self.max_cache_len)
+        prefill = torch.tensor([[[[v] for v in [1.0, 2.0, 0.0, 0.0]]]], dtype=torch.float)
+        static_cache.key_cache[0].copy_(prefill)
+        static_cache.value_cache[0].copy_(prefill)
+        static_cache.update(
+            key_states=torch.tensor([[[[3.0]]]]),
+            value_states=torch.tensor([[[[3.0]]]]),
+            layer_idx=0,
+            cache_kwargs={"cache_position": torch.tensor([2])},
+        )
+        self.assertEqual(
+            static_cache.key_cache[0][0, 0, :, 0].tolist(), [1.0, 2.0, 3.0, 0.0], "StaticCache Scenario 1 failed"
+        )
+
+        # Scenario 2: Fill to capacity
+        static_cache = StaticCache(config=self.config, max_batch_size=1, max_cache_len=self.max_cache_len)
+        prefill = torch.tensor([[[[v] for v in [1.0, 2.0, 3.0, 0.0]]]], dtype=torch.float)
+        static_cache.key_cache[0].copy_(prefill)
+        static_cache.value_cache[0].copy_(prefill)
```

Comment:
Reinitializing the cache (these 4 lines) can be removed, correct? 

(same for the other tests)